tough/src/editor/mod.rs

```diff
@@ -208,6 +259,31 @@ impl RepositoryEditor {
     /// of targets to add, and require advanced performance, you may want to
     /// construct `Target`s directly in parallel and use `add_target()`.
     pub fn add_target_path<P>(&mut self, target_path: P) -> Result<&mut Self>
+    where
+        P: AsRef<Path>,
+    {
+        let (target_name, target) = RepositoryEditor::<T>::build_target(target_path)?;
```

Comment:
I think I prefer this function as it was before. I'm not sure the extra `build_target` function adds much. Or maybe I'm missing something?  Do additional places use this that I missed?

Reply:
Also - this is a good example of why we should get rid of the `<T>` ... it's no fun to always need to specify that every time we call methods